icicle/src/msm/msm.cu

```diff
@@ -726,62 +745,90 @@ namespace msm {
         NUM_BLOCKS = (nof_bms_in_batch + NUM_THREADS - 1) / NUM_THREADS;
         big_triangle_sum_kernel<<<NUM_BLOCKS, NUM_THREADS, 0, stream>>>(buckets, final_results, nof_bms_in_batch, c);
       } else {
+        // the recursive reduction algorithm works with 2 types of reduction that can run on parallel streams
```

Comment:
any chance you move all this reduction logic to a 'reduction_phase()' function? thanks

Reply:
in that case we should also have scalar_splitting_phase() sorting_phase() accumulation_phase() and so on, if we want this refactoring let's do it in a different PR